src/commentscan.l

```diff
@@ -706,15 +712,35 @@ STopt  [^\n@\\]*
                                         }
 <Comment>"<"{SUMMARY}">"                { // start of a .NET XML style brief description
                                           if (!yyextra->HTMLDetails) setOutput(yyscanner,OutputBrief);
-                                          addOutput(yyscanner,yytext);
+                                          else
+                                          {
+                                            int id = yyextra->HTMLDetailsSummary[yyextra->HTMLDetails-1];
+                                            if (id == 0)
+                                            {
+                                              addOutput(yyscanner,"\\htmlonly </p> \\endhtmlonly");
+                                              addOutput(yyscanner,yytext);
+                                            }
+                                            else addOutput(yyscanner,"<p>");
+                                            yyextra->HTMLDetailsSummary.pop_back();
+                                            yyextra->HTMLDetailsSummary.push_back(id+1);
+                                          }
                                         }
 <Comment>"<"{REMARKS}">"                { // start of a .NET XML style detailed description
                                           setOutput(yyscanner,OutputDoc);
                                           addOutput(yyscanner,yytext);
                                         }
 <Comment>"</"{SUMMARY}">"               { // start of a .NET XML style detailed description
-                                          addOutput(yyscanner,yytext);
-                                          if (!yyextra->HTMLDetails) setOutput(yyscanner,OutputDoc);
+                                          if (yyextra->HTMLDetails)
+                                          {
+                                            int id = yyextra->HTMLDetailsSummary[yyextra->HTMLDetails-1];
+                                            if (id == 1)
```

Comment:
what is special about `id==1`? Shouldn't this be `id>0`? I can imagine that 0 is the special in being the first, but 1?

Reply:
I have not been able to create an example where `id == 1` gives problems.
small nested example
```
/** the fie
 <details>
 <summary> The summary 1</summary>
 The details 1


 some
 <summary> The summary 1</summary>
 The details 1
 <details>
 <summary> The summary 1 nested</summary>
 The details 1 nested


 some
 <summary> The summary 1 nested</summary>
 The details 1 nested
 <details>
 <summary> The summary 1 nested 2</summary>
 The details 1 nested 2


 some
 <summary> The summary 1 nested 2</summary>
 The details 1 nested 2
 </details>
 </details>
 </details>
*/
void fie3();
```
works fine with `id == 1` but fails with `id >= 1`.